indra/llconvexdecomposition/llconvexdecompositionvhacd.h

```diff
@@ -0,0 +1,339 @@
+    };
+
+    class VHACDLogger : public VHACD::IVHACD::IUserLogger
+    {
+        void Log(const char* const msg) override
+        {
+            LL_INFOS("VHACD") << msg << LL_ENDL;
+        }
+    };
+
+public:
+
+    LLConvexDecompositionVHACD();
+    virtual ~LLConvexDecompositionVHACD();
+
+    static bool isFunctional();
+    static LLConvexDecomposition* getInstance();
+    static LLCDResult initSystem();
+    static LLCDResult initThread();
+    static LLCDResult quitThread();
+    static LLCDResult quitSystem();
+
+    void genDecomposition(int& decomp);
+    void deleteDecomposition(int decomp);
+    void bindDecomposition(int decomp);
+
+    // Sets *paramsOut to the address of the LLCDParam array and returns
+    // the length of the array
+    int getParameters(const LLCDParam** paramsOut)
+    {
+        *paramsOut = mDecompParams.data();
+        return narrow(mDecompParams.size());
+    }
+
+    int getStages(const LLCDStageData** stagesOut)
+    {
+        *stagesOut = mDecompStages.data();
+        return narrow(mDecompStages.size());
+    }
+
+    // Set a parameter by name. Returns false if out of bounds or unsupported parameter
+    LLCDResult  setParam(const char* name, float val);
+    LLCDResult  setParam(const char* name, int val);
+    LLCDResult  setParam(const char* name, bool val);
+    LLCDResult  setMeshData( const LLCDMeshData* data, bool vertex_based );
+    LLCDResult  registerCallback(int stage, llcdCallbackFunc callback );
+
+    LLCDResult  executeStage(int stage);
+    LLCDResult  buildSingleHull();
+
+    int getNumHullsFromStage(int stage);
+
+    LLCDResult  getHullFromStage( int stage, int hull, LLCDHull* hullOut );
+    LLCDResult  getSingleHull( LLCDHull* hullOut ) ;
+
+    // TODO: Implement lock of some kind to disallow this call if data not yet ready
+    LLCDResult  getMeshFromStage( int stage, int hull, LLCDMeshData* meshDataOut);
+    LLCDResult  getMeshFromHull( LLCDHull* hullIn, LLCDMeshData* meshOut );
+
+    // For visualizing convex hull shapes in the viewer physics shape display
+    LLCDResult generateSingleHullMeshFromMesh( LLCDMeshData* meshIn, LLCDMeshData* meshOut);
+
+    /// Debug
+    void loadMeshData(const char* fileIn, LLCDMeshData** meshDataOut);
+
+private:
+    std::vector<LLCDParam>       mDecompParams;
+    std::array<LLCDStageData, 1> mDecompStages;
+
+    struct LLVHACDMesh
+    {
+        using vertex_type = VHACD::Vertex;
+        using index_type = VHACD::Triangle;
+        using vertex_array_type = std::vector<vertex_type>;
+        using index_array_type = std::vector<index_type>;
+
+        LLVHACDMesh() = default;
+        LLVHACDMesh(const LLCDHull* hullIn)
+        {
+            if (hullIn)
+            {
+                from(hullIn);
+            }
+        };
+
+        LLVHACDMesh(const LLCDMeshData* meshIn, bool vertex_based)
+        {
+            if (meshIn)
+            {
+                from(meshIn, vertex_based);
+            }
+        };
```

Comment:
Semicolons just cuz?
```suggestion
        }

        LLVHACDMesh(const LLCDMeshData* meshIn, bool vertex_based)
        {
            if (meshIn)
            {
                from(meshIn, vertex_based);
            }
        }
```